### PR TITLE
strict mode

### LINKED ISSUE #322: implement "strict" mode
This will be a command line parameter `-strict` which will forcefully disable optional QOL features and enhancements, e.g. colored blood or translucent ghosts monsters, to make sure recorded demos are compliant to speed running rules. This is supposed to be kept in Sync with DSDA-Doom.

### PR DESCRIPTION
**Pull request by rfomin**
Fix #322

### CODE REVIEW

File: Source/hu_stuff.c [hunk 1]
```diff
@@ -1144,7 +1144,8 @@ void HU_Drawer(void)
       }
 
       // [FG] draw player coords widget
-      if (automapactive && !(hud_distributed && automapoverlay) && map_player_coords)
+      if ((automapactive && STRICTMODE(map_player_coords) == 1)
```
Comment (fabiangreffrath) [inline Source/hu_stuff.c:1147]: How about the `hud_distributed` condition?

File: Source/m_menu.c [hunk 2]
```diff
@@ -3710,6 +3713,37 @@ void static M_SmoothLight(void)
   P_SegLengths(true);
 }
 
+static void M_UpdateStrictModeItems(void)
+{
+  if (strictmode)
+  {
```
Comment (fabiangreffrath) [inline Source/m_menu.c]: How about `S_STRICT` that acts like `S_DISABLE` but only in strict mode?

File: Source/hu_stuff.c (hunk 1 above)
Comment (rfomin) [reply Source/hu_stuff.c]: Fixed in https://github.com/fabiangreffrath/woof/pull/597/commits/0b742ae86ed429ac0fbf683e73edf0b8a25bab2b

File: Source/m_menu.c (hunk 2 above)
Comment (rfomin) [reply Source/m_menu.c]: I don't know why this should help?

Comment (fabiangreffrath) [reply Source/m_menu.c]: `if (flags & S_DISABLE || STRICTMODE(flags & S_STRICT))`

Comment (rfomin) [reply Source/m_menu.c]: I like `M_UpdateStrictModeItems` better because we have all the menu options in one place.

Comment (fabiangreffrath) [reply Source/m_menu.c]: Okay, fine with me.

**Comment by rfomin** [comment]
Brightmaps is cheating probably. Centered weapon is restricted, but you can disable bobbing entirely in DSDA-Doom :confused:

**Comment by fabiangreffrath** [comment]
> 1Centered weapon is restricted, but you can disable bobbing entirely in DSDA-Doom 😕

Hm, are any of these documented in the demo header?

**Comment by rfomin** [comment]
> Hm, are any of these documented in the demo header?

No, it's cosmetic bobbing.

**Comment by kraflab** [comment]
In dsda-doom, bobbing is still "on" in cl9+ headers, but it can be visually disabled (decoupled visuals from the player data). As far as why this is allowed, it's specifically because some people can't play the game with bobbing on for health reasons.

**Comment by rfomin** [comment]
@fabiangreffrath I believe both this and #596 are done.

File: Source/m_menu.c [hunk 3]
```diff
@@ -3094,10 +3098,7 @@ static const char *weapon_attack_alignment_strings[] = {
 static void M_UpdateCenteredWeaponItem(void)
 {
   // weap_center
-  if (cosmetic_bobbing)
-      weap_settings1[12].m_flags &= ~S_DISABLE;
-  else
-      weap_settings1[12].m_flags |= S_DISABLE;
+  DISABLE_ITEM(!cosmetic_bobbing, weap_settings1[13]);
```
Comment (fabiangreffrath) [inline Source/m_menu.c:3101]: `[12]` or `[13]`?

File: Source/m_menu.c [hunk 4]
```diff
@@ -3220,27 +3221,11 @@ setup_menu_t stat_settings1[] =  // Status Bar and HUD Settings screen
...
-        }
-    }
-    else
-    {
-        stat_settings2[8].m_flags  |= S_DISABLE;
-        stat_settings2[9].m_flags |= S_DISABLE;
-        stat_settings2[10].m_flags |= S_DISABLE;
-        stat_settings2[11].m_flags |= S_DISABLE;
-    }
+    DISABLE_ITEM(!hud_crosshair, stat_settings2[8]);
```
Comment (fabiangreffrath) [inline Source/m_menu.c:3224]: So, crosshair per-se is allowed in strict mode?

File: Source/m_menu.c (hunk 3 above)
Comment (rfomin) [reply Source/m_menu.c:3101]: `[13]`, you can turn off bobbing and enable recoil pitch, but not center weapon. Not sure why, maybe we should revist that code.

File: Source/m_menu.c (hunk 4 above)
Comment (rfomin) [reply Source/m_menu.c:3224]: Yes, crosshairs are allowed (you can physically draw a dot in the center of the screen).

**Comment by fabiangreffrath** [comment]
This was ready to merge, right?

**Comment by rfomin** [comment]
> This was ready to merge, right?

Yes I think so.

**Pull request merged**